octoprint_mrbeam/software_update_information.py

```diff
@@ -100,231 +215,244 @@ def switch_software_channel(plugin, channel):
+            _logger.error("Error while setting octoprint update config")
+
+        for module_id, module in update_info.items():
+            try:
+                if tier in [
+                    SW_UPDATE_TIER_BETA,
+                    SW_UPDATE_TIER_DEV,
+                    SW_UPDATE_TIER_PROD,
+                ]:
+                    if _is_override_in_settings(
+                        plugin, module_id
+                    ):  # check if update config gets overriden in the settings
+                        return
+                    sw_update_config[module_id] = {}
+                    moduleconfig = sw_update_config[module_id]
+                    moduleconfig.update(defaultsettings)
+
+                    # get update info for tier branch
+                    tierversion = get_tier_by_id(tier)
+
+                    if tierversion in moduleconfig:
+                        module_tier = moduleconfig[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # set tier config from default settings
+                    if tierversion in module:
+                        module_tier = module[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # override tier config from tiers set in config_file
+
+                    _set_update_config_from_dict(
+                        moduleconfig, module
+                    )  # set default config from file for module
+
+                    # have to be after the default config from file
+                    if "beamos_date" in module:
+                        beamos_date_config = module["beamos_date"]
+                        for date, beamos_config in beamos_date_config.items():
+                            _logger.debug(
+                                "date compare %s >= %s -> %s",
+                                beamos_date,
+                                datetime.strptime(date, "%Y-%m-%d").date(),
+                                beamos_config,
+                            )
+                            if (
+                                beamos_date
+                                >= datetime.strptime(date, "%Y-%m-%d").date()
+                            ):
+                                if tierversion in beamos_config:
+                                    beamos_config_module_tier = beamos_config[
+                                        tierversion
+                                    ]
+                                    _set_update_config_from_dict(
+                                        beamos_config, beamos_config_module_tier
+                                    )  # override tier config from tiers set in config_file
+                                _set_update_config_from_dict(
+                                    moduleconfig, beamos_config
+                                )
```

Comment:
This is where a deep merge using `dict_merge` is nice; I think it can replace this extra `if` statement